Cython/Utils.py

```diff
@@ -305,7 +305,7 @@ def find_versioned_file(directory, filename, suffix,
     assert not suffix or suffix[:1] == '.'
     path_prefix = os.path.join(directory, filename)
 
-    matching_files = glob.glob(path_prefix + ".cython-*" + suffix)
+    matching_files = glob.glob(''.join([ '['+c+']' if c in '[]*?' else c for c in path_prefix]) + ".cython-*" + suffix)
```

Comment:
You can use `glob.escape()` for this. It's available since Py3.4, the master branch requires Py3.7+.

Reply:
That was my first thought, but https://cython.org says "The C code is generated once and then compiles with all major C/C++ compilers in CPython 2.6, 2.7 (2.4+ with Cython 0.20.x) as well as 3.5 and all later versions" so I assumed I had to be backwards compatible with 2.6 and 2.7, but thanks for clarifying. I did see something in the source code about 3.7+ so I had conflicting information. I will fix this, thank you!

Reply:
The website is a little outdated now in terms of what Cython supports.

The master branch is working towards a release that'll drop support for a lot of older versions but that release hasn't happened yet so it isn't _too_ misleading (yet)

Reply:
@da-woods So is it best to leave it as-is for now? We are eager to have this fix sooner than later because it breaks our build. Perhaps I can file a follow-on ticket to clean it up for the strict 3.7+ release and leave it as technical debt?

Reply:
Ah, got it. So `master` → 3.1, not 3.0.9. How does the backport to 3.0.9 procedure work — create a separate branch and issue a pull request into the `3.0.x` branch? Probably better to tag a branch at the current commit and pull from there. Thanks for the feedback; I'll get right on it.